Classes/Configuration/AutomaticConfigurator.php

```diff
@@ -83,7 +83,7 @@ protected function addLanguages(array &$configuration) {
 		if ($this->hasStaticInfoTables) {
 			$languages = $this->databaseConnection->exec_SELECTgetRows('t1.uid AS uid,t2.lg_iso_2 AS lg_iso_2', 'sys_language t1, static_languages t2', 't2.uid=t1.static_lang_isocode AND t1.hidden=0');
 		} else {
-			$languages = $this->databaseConnection->exec_SELECTgetRows('t1.uid AS uid,t1.uid AS lg_iso_2', 'sys_language t1', 't1.hidden=0');
+			$languages = $this->databaseConnection->exec_SELECTgetRows('t1.uid AS uid,t1.language_isocode AS lg_iso_2', 'sys_language t1', 't1.hidden=0');
```

Comment:
The whole logic should be changed to:

If CMS 7, use `language_isocode` unless empty, otherwise fall back to 6.2 handling.
If CMS 6.2, if SIT is available and `static_lang_isocode` is set, resolve the relation.